Docs/doc 850 get started info updates (#842)
spatialos/UnrealGDK: SpatialGDK/Documentation/content/get-started/gdk-template.md

<!---
TODO: add links from deployments to concepts
-->

When you launch a deployment, SpatialOS sets up the world based on a [snapshot]({{urlRoot}}/content/spatialos-concepts/generating-a-snapshot), then starts up the worker instances needed to run the game world.

<!---
TODO: add links from workers to concepts
-->

You'll find out more about schema, snapshots and workers later on in this tutorial. 

#### Deploy locally with multiple clients

Before you launch a deployment (local or cloud) you must generate [schema]({{urlRoot}}/content/spatialos-concepts/schema) and a [snapshot]({{urlRoot}}/content/spatialos-concepts/generating-a-snapshot). 

1. In the Unreal Editor, on the GDK toolbar, select **Schema** to generate schema.<br/>
![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)<br/>
_Image: On the GDK toolbar in the Unreal Editor select **Schema**_<br/>
1. Select **Snapshot** to generate a snapshot.<br/>
![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/snapshot-button.png)<br/>
_Image: On the GDK toolbar in the Unreal Editor select **Snapshot**_<br/>

> <%(#Expandable title="What is schema?")%>Schema is where you define all the components in your world. Schema is defined in .schema files, written in schemalang, which is a wrapper around Google’s protobuf.

Author: Nice! BTW did you mean to delete "in" before ".schema files"? If not, could you add it back in (so I can commit the change)? Thanks!

> <%(#Expandable title="What is schema?")%>Schema is where you define all the components in your world. Schema is defined in .schema files, written in schemalang, which is a wrapper around Google’s protobuf.

Reviewer: Mistake. Added back in. Committing.

![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/start-button.png)<br/>
_Image: On the GDK toolbar in the Unreal Editor select **Start**_<br/>
1. On the Unreal Editor toolbar, open the **Play** drop-down menu.
1. Under **Modes**, select **New Editor Window (PIE)**.<br/>
1. Under **Multiplayer Options**, set the number of players to **2** and ensure that the check box next to **Run Dedicated Server** is checked. (If it is unchecked, select the checkbox to enable it.)<br/>
![]({{assetRoot}}assets/set-up-template/template-multiplayer-options.png)<br/>
_Image: The Unreal Engine **Play** drop-down menu, with **Multiplayer Options** and **New Editor Window (PIE)** highlighted_<br/>
1. On the Unreal Engine toolbar, select **Play** to run the game, and you should see two clients start.<br/><br/>
![]({{assetRoot}}assets/set-up-template/template-two-clients.png)<br/>
_Image: Two clients running in the editor, with player Actors replicated by SpatialOS and the GDK_<br/>

> 1. Open the Inspector. You should see that a local SpatialOS deployment is running with one server-worker instance and two client-worker instances connected. You can also find and follow around the two player entities.<br/><br/>

Reviewer: We don't say what the Inspector is or how to open it.
```suggestion
1. Open the Inspector using the local URL you were given above: `http://localhost:21000/inspector`.</br>  You should see that a local SpatialOS deployment is running with one server-worker instance and two client-worker instances connected. You can also find and follow around the two player entities.<br/><br/>
```

![]({{assetRoot}}assets/set-up-template/template-two-client-inspector.png)<br/>
_Image: The Inspector showing the state of your local deployment_<br/>
1. When you're done, select **Stop** in the GDK toolbar to stop your local SpatialOS deployment.<br/>![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/stop-button.png)<br/>
_Image: On the GDK toolbar in the Unreal Editor select **Stop**_<br/>

> 

Reviewer: We don't really talk about what the Inspector is. I know it's not vital but 

```suggestion
<%(#Expandable title="What is the Inspector?")%> The Inspector is a web-based tool that you use to explore the internal state of a game's SpatialOS world. It gives you a real-time view of what’s happening in a local or cloud deployment. <br/>
The Inspector we are using here is looking at a local deployment, so it's a local URL with the Inspector running locally. When running locally, the Inspector automatically downloads and caches the latest Inspector client from the internet. When you use the Inspector in a cloud deployment, you access the Inspector through the Console via the URL https://console.improbable.io.

You can find out more about the [Inspector]({{urlRoot}}/content/glossary#inspector) and the [Console]({{urlRoot}}/content/glossary#console) in the _Glossary_. <%(/Expandable)%> 
```

##### Step 1: Set up your SpatialOS project name. 
When you signed up for SpatialOS, your account was automatically associated with an organisation and a project, both of which have the same generated name.

1. Find this name by going to the [Console](https://console.improbable.io). 
The name should look something like `beta_randomword_anotherword_randomnumber`. In the example below, it’s `beta_yankee_hawaii_621`. <br/>![Toolbar]({{assetRoot}}assets/set-up-template/template-project-page.png)<br/>_Image: The SpatialOS Console with a project name highlighted._
2. In File Explorer, navigate to the `<YourProject>/spatial` directory and open the `spatialos.json` file in a text editor of your choice.
3. Replace the `name` field with the project name shown in the Console. This tells SpatialOS which SpatialOS project you intend to upload to.

> 

Reviewer: We don't say what the Console is. Suggest: 

```suggestion
<%(#Expandable title="What is the Console?")%> The Console is a web-based tool for managing cloud deployments. You gives you access to information about your games' SpatialOS project names, the SpatialOS assemblies you have uploaded, the internal state of any games you have running (via the Inspector), as well as logs and metrics. <br/>
You can find out more about the [Console]({{urlRoot}}/content/glossary#console) in the _Glossary_. <%(/Expandable)%>
```

##### Step 2: Build your worker assemblies

> Workers are the programs that connect to a SpatialOS world. In general, you use server-worker instances simulate the world, and players use client-worker instances to connect to your game. Worker assemblies are `.zip` files that contain all the files that your game uses when running in the cloud.

Reviewer: I think "connect" is confusing as that suggests they are networking when they aren't really about networking.
By saying "you use" and then "players use", it's a bit confusing as to who is doing what. As you (devs) use in terms of doing development but players use in terms of playing a game. 
Also, we are saying servers simulate a world but players connect to a game. The game and world are the same thing here, so we should use the same word.

```suggestion
Workers are the programs that compute a SpatialOS world. In general, server-worker instances simulate the world, while game players connect to the world through client-worker instances. Worker assemblies are `.zip` files that contain built-out workers; that is all the files with compiled code that your game uses for running in the cloud.
```

<!---
TODO: add link from server-workers to concepts
-->

<!--<%(#Expandable title="More information about worker assemblies")%>More info<%(/Expandable)%>-->

> **Note:** In the following commands, you must replace **`YourProject`** with the name of your Unreal project (chosen when you created the project from the Unreal Editor).

Reviewer: ```suggestion
**Note:** In the following commands, you must replace **`YourProject`** with the name of your Unreal project (the one you chose when you created the project from the Unreal Editor - not the name of your SpatialOS project).
```

1. In a terminal window, navigate to your `<ProjectRoot>` directory.
2. Build a server-worker assembly by running the following command: 
```
Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat YourProject Linux Development YourProject.uproject
```
3. Build a client-worker assembly by running the following command: 
```
Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat YourProject Win64 Development YourProject.uproject
```

> Alternatively you can use the `BuildProject.bat` script found in the `<ProjectRoot>` directory to run both of these commands automatically. 

Reviewer: I suggest introducing it as two options from the top - using a `.bat` script to run these commands automatically or running commands manually.
